sonar-orchestrator/src/test/java/com/sonar/orchestrator/OrchestratorBuilderTest.java

```diff
@@ -22,37 +22,55 @@
 import com.sonar.orchestrator.config.Configuration;
 import com.sonar.orchestrator.locator.MavenLocation;
 import com.sonar.orchestrator.locator.PluginLocation;
+import java.io.File;
 import java.net.URL;
 import java.util.NoSuchElementException;
 import org.junit.BeforeClass;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class OrchestratorBuilderTest {
 
+  private static URL updateCenterUrl;
+
   @Rule
   public ExpectedException expectedException = ExpectedException.none();
-  private static URL updateCenterUrl;
+
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
 
   @BeforeClass
   public static void prepare() {
     updateCenterUrl = OrchestratorBuilderTest.class.getResource("/update-center-test.properties");
   }
 
   @Test
-  public void sonarVersionIsMandatory() {
+  public void throw_IAE_if_both_zip_path_and_version_of_sonarqube_are_missing() {
     expectedException.expect(IllegalStateException.class);
-    expectedException.expectMessage("Missing SonarQube version");
+    expectedException.expectMessage("Version or path to ZIP of SonarQube is missing");
 
     new OrchestratorBuilder(Configuration.create())
       .setOrchestratorProperty("orchestrator.updateCenterUrl", updateCenterUrl.toString())
       .build();
   }
 
   @Test
+  public void install_sonarqube_from_local_zip_file() throws Exception {
+    File zip = temp.newFile("sonarqube-6.3.1.4567.zip");
+
+    // version is not set
+    Orchestrator orch = new OrchestratorBuilder(Configuration.create())
+      .setZipFile(zip)
+      .build();
+
+    assertThat(orch.getDistribution().getZipFile().get()).isEqualTo(zip);
```

Comment:
in fact no. You're talking about `Server#version()`, not `Distribution#version()`. The first one is available only if orchestrator is started. That is not the case in this test. 
Note that verification of version is done in `OrchestratorBuilderLongTest` (kind of IT).